be/src/exec/analytor.cpp

```diff
@@ -140,9 +140,10 @@ Status Analytor::prepare(RuntimeState* state, ObjectPool* pool, RuntimeProfile*
             _need_partition_materializing = true;
         }
 
-        if (fn.name.function_name == "cume_dist") {
+        if (fn.name.function_name == "cume_dist" || fn.name.function_name == "percent_rank") {
             if (!state->enable_pipeline_engine()) {
-                return Status::NotSupported("The CUME_DIST window function is only supported by the pipeline engine.");
+                return Status::NotSupported(strings::Substitute(
+                        "The $0 window function is only supported by the pipeline engine.", fn.name.function_name));
             }
```

Comment:
it is not very clear
so can we introduce a function "should_set_partition_size" to tacke with functions like percent_rank and dist_cume?

Reply:
Done, I use `require_partition_size(fn.name.function_name)` to tackle with functions like percent_rank and dist_cume. `_should_set_partition_size` to indicate whether partition size needs to be set.
PTAL.